lib/serve/functions.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+var _ = require('lodash');
+var chalk = require('chalk');
+var path = require('path');
+var RSVP = require('rsvp');
+var getProjectId = require('../getProjectId');
+var logger = require('../logger');
+var utils = require('../utils');
+var parseTriggers = require('../parseTriggers');
+var EmulatorController;
+var controller;
+
+function _pollOperation(service, operation) {
+  return new RSVP.Promise(function(resolve, reject) {
+    var poll = function() {
+      service.operations.get({
+        name: operation[0].name
+      }, function(err, _operation) {
+        if (err) {
+          reject(err);
+        } else if (_operation.done) {
+        // TODO: handle deploy failures
+          resolve(_operation.response.value);
+        } else {
+          setTimeout(poll, 500);
+        }
+      });
+    };
+    poll();
+  });
+}
+
+function _startEmulator(options) {
+  try {
+    EmulatorController = require('@google-cloud/functions-emulator/src/cli/controller');
+  } catch (err) {
+    var msg = err;
+    if (process.version !== 'v6.9.1') {
```

Comment:
I don't think we should be requiring an exact version of Node. It should probably be >= whatever GCF is running. This will be a pretty annoying burden on devs otherwise.